### ISSUE TITLE
extend filter syntax to handle title lists, not title sets

### ISSUE THREAD

**Issue by tobibeer** [Jermolene/TiddlyWiki5#2097]
The current filter mechanism joins output titles to the input forming a set (of unique titles), not a list (that would allow duplicates).

To allow working with lists (containing duplicates) amend the **general filter syntax** such that filter runs can be joined in either set-mode (default) or list-mode by using a new filter run prefix, e.g. `&`.

currently => **set-mode**:

```
{{{ foo foo foo }}}
```

> [foo](http://tiddlywiki.com)

vs. => **list-mode**:

```
{{{ foo &[[foo]] &[[foo]] }}}
```

> [foo](http://tiddlywiki.com) [foo](http://tiddlywiki.com) [foo](http://tiddlywiki.com)

The `&` as a prefix for a filter run instructs TiddlyWiki to append the results to the input titles while keeping the input titles, e.g:

```
{{{ [foo[bar]] &[operator:sufix[operand]] }}}
```
## Use Case

A user may wish to store and handle routing instructions in a list field, e.g.:

```
go: north north north west north west west south
```
## Considerations

In set mode `-` should remove all instances (duplicates):

```
[foo[bar]] -[any-filter[stuff]]
```

For a different behavior a dedicated filter is needed.

**Comment by inmysocks** [comment]
This could be done by introducing a new prefix other than + and - that means allow duplicates. The test for duplicates happens in the compileFilter function in the same place that the prefixes + and - are handled so it would just be another section in the case statement there, so something like d[foo[bar]]. To avoid confusion, having - remove all instances would be a good idea because the operators introduced along with action-listops cover the instances where single items are removed.

**Comment by tobibeer** [comment]
@inmysocks, makes sense. I have updated the issue description accordingly, however proposing `&`, instead of `d` to look more like an operator.

I have also removed the proposal to flag an entire filter string to be treated in "list-mode". I think it unnecessarily complicates things.

I guess `-[foo[bar]]` should always remove all occurrences and — if we wanted to remove a particular title only — we needed one of the listops filters from #2037.
